tools/topology/platform/intel/intel-generic-bt.m4

```diff
@@ -0,0 +1,78 @@
+#
+# BT audio offset support
+#
+
+include(`ssp.m4')
+
+# define default SSP port
+ifdef(`SSP_INDEX',`',
+`define(SSP_INDEX, `2')')
+
+define(`SCO_MCLK', 19200000)
+define(`SSP_NAME', concat(concat(`SSP', SSP_INDEX),`-BT'))
+
+# variable that need to be defined in upper m4
+ifdef(`BT_PIPELINE_PB_ID',`',`fatal_error(note: Need to define playback pcm id for intel-generic-bt
+)')
+ifdef(`BT_PIPELINE_CP_ID',`',`fatal_error(note: Need to define capture pcm id for intel-generic-bt
+)')
+ifdef(`BT_PIPELINE_BE_ID',`',`fatal_error(note: Need to define pipeline link id for intel-generic-bt
+)')
+ifdef(`HW_CONFIG_ID',`',`fatal_error(note: Need to define hw_conf_id for intel-generic-bt
+)')
+
+define(`CHANNELS_MIN', 1)
+
+# BT offload playback
+PIPELINE_PCM_ADD(sof/pipe-passthrough-playback.m4,
+	BT_PIPELINE_PB_ID, 50, 2, s16le,
+	1000, 0, 0,
+	8000, 16000, 48000)
```

Comment:
Should be 	8000, **48000**, 48000)

Reply:
the macro seems to have issues here. It's intended for a min 8 and max 48 but will all intermediate frequencies in between, i.e a range.
Here we would want to have a list of 8, 16 and 48 supported, not a range.

@ranj063 @juimonen can you chime in?

Reply:
this is how it is currently in asoc/alsa also... I think. I guess there's no guarantee that all possible sample rates between a "range" will work? and what are the possible rates? what I mean that if we have a pcm range from 8 to 48 -> does it guarantee that for e.g. that 22050kHz to 48khz works?

So I'm all in for specifying just the supported rates, just not sure alsa supports it.

Reply:
still to clarify, the above macro was originally way back meant for enabling the SRC on the pipeline. So the idea was to say that in the src component we accept the range like from 8khz to 48khz. Then the last param was the "fixed" dai rate. it is not even used for anything else than SRC configuration. So we could for example have definition (8000, 96000, 48000) -> SRC would understand it needs to convert the range 8-96khz to 48. First 2 parameters are set for pcm min max rate, so that alsa let's these streams even go the dsp.

Now when we start to have more dynamic pipeline, this last rate on the definition doesn't make too much sense and we should/could get the dai rate from params. I'm not sure still is it possible for the SRC component to get the dai params in stream start -> reconfigure itself. The pipeline walk should be checked for this. Way back you could not "sniff" the dai rate from SRC. it might have changed.